libcxx/include/string

```diff
@@ -748,6 +748,41 @@ struct __can_be_converted_to_string_view
 struct __uninitialized_size_tag {};
 struct __init_with_sentinel_tag {};
 
+#ifdef _LIBCPP_ABI_ALTERNATE_STRING_LAYOUT
+template <class _CharT, size_t __min_cap, size_t _Padding = sizeof(_CharT) - 1>
+struct __short_layout_alternate {
+  _CharT __data_[__min_cap];
+  unsigned char __padding_[_Padding];
```

Comment:
Can you try out @philnik777 's suggestion for the padding? I think it should work with our current CI setup.